libopenarc/arc.c

```diff
@@ -3560,7 +3560,57 @@ arc_get_domain(ARC_MESSAGE *msg)
 */
 
 const char *
-arc_chain_str(ARC_MESSAGE *msg)
+arc_chain_status_str(ARC_MESSAGE *msg)
 {
 	return arc_code_to_name(chainstatus, msg->arc_cstate);
 }
+
+/*
+**	ARC_CHAIN_CUSTODY_STR -- retrieve domain chain, as a string
+**
+**	Parameters:
+**      msg -- ARC_MESSAGE object
+**      buf -- where to write
+**      buflen -- bytes at "buf"
+**
+**	Return value:
+**	    Number of bytes written
+*/
+int
+arc_chain_custody_str(ARC_MESSAGE *msg, u_char *buf, size_t buflen)
+{
+	u_int set;
+	u_char *inst;
+	ARC_KVSET *kvset;
+	char *str = NULL;
+	char *delim = "";
```

Comment:
This isn't needed because it never changes.  You can just pass "" literally to the call below.

Reply:
It does change. Subsequent arc.chain results are delimited by a colon.

Reply:
I think I get it now. I can just use dstring and arc_dstring_printf. Only problem is there is a weird dependency: arc-types.h includes arc.h. That doesn't seem to make sense. I can see arc.h including arc-types.h...